pkg/rpc/server.go

```diff
@@ -225,6 +227,38 @@ Methods:
 		} else {
 			results = "Invalid public account address"
 		}
+	case "getrawtransaction":
+		var err error
+
+		param0, errParam := reqParams.ValueWithType(0, "string")
+		if errParam != nil {
+			err = errors.Wrap(errParam, "expected param at index 0 to be a valid string transactionID")
+			resultsErr = NewInvalidParamsError(err.Error(), err)
+		} else if txHash, err := util.Uint256DecodeString(param0.StringVal); err != nil {
+			err = errors.Wrapf(err, "param at index 0, (%s), could not be decode to Uint256", param0.StringVal)
+			resultsErr = NewInvalidParamsError(err.Error(), err)
+		} else if tx, height, err := s.chain.GetTransaction(txHash); err != nil {
+			err = errors.Wrapf(err, "Invalid transaction hash: %s", txHash)
+			resultsErr = NewInvalidParamsError(err.Error(), err)
+		} else if len(reqParams) >= 2 {
+			_header := s.chain.GetHeaderHash(int(height))
+			header, err := s.chain.GetHeader(_header)
+
+			param1, exists := reqParams.ValueAtAndType(1, "number")
+			if !exists {
+				err = errors.New("expected param at index 1 to be either 1 or 0")
+				resultsErr = NewInvalidParamsError(err.Error(), err)
+			} else if verboseFlag := param1.IntVal; verboseFlag != 0 && verboseFlag != 1 {
```

Comment:
this is not equivalent to that is in CSharp code

Reply:
Yes, you are right. C# code is more weak here in the sense that  `verbose`  can be any number. If it is 0 then it is interpreted as FALSE . Otherwise everything else is TRUE. In that regards there is a discrepancy with the [documentation] (https://docs.neo.org/en-us/node/cli/latest-version/api/getrawtransaction.html
) which only mention two options: 0 or 1:

>Parameter Description
Txid: Transaction ID

Verbose: Optional, the default value of verbose is 0. When verbose is 0, the serialized information of the block is returned, represented by a hexadecimal string. If you need to get detailed information, you will need to use the SDK for deserialization. When verbose is 1, detailed information of the corresponding block in Json format string, is returned.

 If you like I can follow strictly the C# code and neglect the documentation.

Reply:
https://github.com/neo-project/neo-cli/issues/300

I have created an issue on that, but I believe that there will be no changes. So we have to match the C# code for now